pkgs/tools/security/ghidra/with-extensions.nix

```diff
@@ -0,0 +1,29 @@
+{ lib
+, symlinkJoin
+, makeBinaryWrapper
+, ghidra
+, ghidra-extensions
+}:
+
+let
+  allExtensions = lib.filterAttrs (n: pkg: lib.isDerivation pkg) ghidra-extensions;
+
+  /* Make Ghidra with additional extensions
+     Example:
+       pkgs.ghidra.withExtensions (p: with p; [
+         ghostrings
+       ]);
+       => /nix/store/3yn0rbnz5mbrxf0x70jbjq73wgkszr5c-ghidra-with-extensions-10.2.2
+  */
+  withExtensions = f: (symlinkJoin {
+    name = "ghidra-with-extensions-${lib.getVersion ghidra}";
+    paths = [ ghidra ] ++ (f allExtensions);
```

Comment:
Sorry I was a bit unclear. My point was that with the change to add the extensions to `findDefaultApplicationRootDirs`, we should be able to remove `ghidra` from this symlinkJoin. Ghidra will look for extensions in both its home directory and the NIX_GHIDRAHOME, so there's no need to symlink the entire ghidra here.

Reply:
Okay, if, I understood you correctly, the new version should do what you want?